### ISSUE TITLE
Mouse motion event not captured on wxListCtrl

### ISSUE THREAD

**Issue by mcastel1** [wxWidgets/wxWidgets#24797]
### Description
Please use the following sections in your report:

#### Bug description:

Mouse motion events are not captured by wxListCtrl

#### Expected vs observed behaviour:

When the mouse hovers over the wxListCtrl and moves, the method OnMouseMovement, which must be triggered by a wxEVT_MOTION, is not called, see the sample code. 

#### Patch or snippet allowing to reproduce the problem:

The problem can be reproduced by adding a few lines to the listctrl sample, see
 the files listtest_modified.cpp and listtest_modified.h attached: [files.zip](https://github.com/user-attachments/files/16880228/files.zip). 
 These files differ by listtest.cpp and listtest.h by the addition of the OnMouseMovement method:, bound to wxEVT_MOTION:  

```
 % diff listtest_modified.cpplistctrl/listtest.cpp
325,328d324
< 
< void MyFrame::OnMouseMovement(wxMouseEvent& event){
< 
<     std::cout << "OnMouseMovement has been called" << std::endl;
330,332d325
<     Close(true);
< }
< 
470,471d462
<         m_listCtrl->wxListCtrl::Bind(wxEVT_MOTION, &MyFrame::OnMouseMovement, this);
< 
```

```
% diff listtest_modified.h listctrl/listtest.h 
102d101
<     
107d105
<     void OnMouseMovement(wxMouseEvent&);
```

#### To Reproduce:

1. Run the listcontrol sample with listtest_modified.cpp and listtest_modified.h
2. Move the mouse on the listcontrol 
3. See that the method MyFrame::OnMouseMovement is not called

### Platform and version information

- wxWidgets version  you use: 3.2.5
- wxWidgets port  you use:
- OS  and its version: OSx 14.6
  
  + GTK version: 
  + Which GDK  backend is used:
  + Desktop environment :
  + Current theme:

**Comment by vadz** [comment]
We probably need to provide `GetMainWindow()` to allow binding to the events in `m_mainWin` of the generic `wxListCtrl` used in wxOSX. For now you can use it directly using the appropriate conditional compilation directives, but note that, generally speaking, you're not supposed to be able to hook into low-level keyboard/mouse events in the native controls, only higher level events such as `wxEVT_LIST_ITEM_FOCUSED` etc should be used.

Also please read our [bug reporting guidelines](https://www.wxwidgets.org/develop/how-to-report-issues/), in particular the part about not attaching ZIP files and using unified diffs. TIA!

**Comment by mcastel1** [comment]
" For now you can use it directly using the appropriate conditional compilation directives" ... I don't understand anything of this. How may I get the wxMouseEvent to be captured with the current version of wxWidgets ? 
As for " you're not supposed to be able to hook into low-level keyboard/mouse events in the native controls", how can I do if I need a mouve movement event to be captures on a wxListCtrl then ? 
Thanks

**Comment by oneeyeman1** [comment]
@mcastel1 ,

1. Instead of m_listCtrl use GetMainWindow().

2. Why do you need that? What problem you are trying to solve or what you want to achieve?

Thank you.

**Comment by vadz** [comment]
> " For now you can use it directly using the appropriate conditional compilation directives" ... I don't understand anything of this. How may I get the wxMouseEvent to be captured with the current version of wxWidgets ? 

You may bind to `wxListCtrl::m_mainWin`. You'll need to do it inside `#ifndef __WXMSW__` (and use `wxListCtrl` itself otherwise) as `m_mainWin` doesn't exist in the wxMSW version of this control.

> As for " you're not supposed to be able to hook into low-level keyboard/mouse events in the native controls", how can I do if I need a mouve movement event to be captures on a wxListCtrl then ? Thanks

This is not something officially supported. It may work if you do what I wrote above, but, in general, we just can't always get the low-level events from native controls.

**Comment by mcastel1** [comment]
I tried all your suggestions, nothing works:

` (m_listCtrl->wxListCtrl::m_mainWin)->Bind(wxEVT_MOTION, &MyFrame::OnMouseMovement, this);`
`GetMainWindow()->Bind(wxEVT_MOTION, &MyFrame::OnMouseMovement, this);`

**Comment by andykt** [comment]
This sounds like the regression bug caused by the addition of m_osxClipView in wxOSX 3.2.5.  I have the same problem with wxHtmlWindow not seeing hover events (see #24790).  Until the bug is fixed, the only solution I've found is to comment out the code in wxWidgetCocoaImpl::UseClippingView (in src/osx/cocoa/window.mm) that sets m_osxClipView.

**Comment by vadz** [comment]
@mcastel1 Please check if this problem is fixed by 568bea42876828e52e11b6bc1293e83683eb6073.

**Comment by mcastel1** [comment]
It is, thank you ! :)
